synapse_data_warehouse/synapse_raw/tables/V2.23.2__refresh_node_snapshot.sql

```diff
@@ -3,54 +3,77 @@ USE ROLE sysadmin;
 USE SCHEMA {{database_name}}.synapse_raw; --noqa: JJ01,PRS,TMP,CP02
 USE WAREHOUSE compute_medium;
 
--- Drop all records from this table
-TRUNCATE TABLE nodesnapshots;
+SET num_records_to_be_reloaded = (
+    SELECT count(*)
+    FROM nodesnapshots
+    WHERE snapshot_date >= '2024-11-01'
+);
```

Comment:
From the script I saw they paginate over the 50k versions and get the most recent 50k versions.

It might be worthwhile to confirm what the behavior is here. This is the line of code I was referring to:

https://github.com/Sage-Bionetworks/Synapse-Repository-Services/blob/ea0fa386da6dda0a3760b097b92883facc1c2795/services/workers/src/main/java/org/sagebionetworks/snapshot/workers/writers/NodeObjectRecordWriter.java#L120

Reply:
@BryanFauble This feels more like a potential upstream data warehouse issue. It doesn't make sense to me at this point for us to try to preserve history which our upstream process is intentionally not collecting. And, you're right, this is an extreme edge case that would take a confluence of unlikely factors to run into, or even have an impact on downstream analysis.

Reply:
The max version number is 36333, so perhaps this is a case where we "worry about it when we get there".  

Feel free to merge if you're comfortable with this PR @philerooski !  Thanks!